frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -104,297 +178,465 @@ const defaultValueFor = (field: OperandField) => {
+// maintain. The structure of the form field state should be redesigned so that it is one structure
+// that more closely matches the nested nature of the form itself.
+// We should also break this down into smaller components and dispatch actions from those
+// components to update parent state.
 export const CreateOperandForm: React.FC<CreateOperandFormProps> = ({
+  buffer,
   clusterServiceVersion,
   openAPI,
   operandModel,
   providedAPI,
   namespace,
-  onChange,
-  sample,
+  onToggleEditMethod,
 }) => {
-  const fieldsFromProps: OperandField[] = React.useMemo(
-    () =>
-      (!_.isEmpty(clusterServiceVersion && providedAPI.specDescriptors)
-        ? fieldsFor(providedAPI)
-        : fieldsForOpenAPI(openAPI)
-      )
-        .map((field) => {
-          const capabilities = field.capabilities || [];
-          const openAPIProperties = _.get(openAPI, 'properties.spec.properties');
-          if (_.isEmpty(openAPIProperties)) {
-            return { ...field, capabilities };
-          }
-          const schemaPath = field.path.split('.').join('.properties.');
-          const required = (_.get(
-            openAPI?.properties,
-            _.dropRight(['spec', ...field.path.split('.')])
-              .join('.properties.')
-              .concat('.required'),
-            [],
-          ) as string[]).includes(_.last(field.path.split('.')));
-          const type = _.get(openAPIProperties, schemaPath.concat('.type')) as JSONSchema6TypeName;
-          const validation = _.pick(openAPIProperties[schemaPath], [
-            ...Object.keys(Validations),
-          ]) as OperandField['validation'];
-
-          return { ...field, capabilities, type, required, validation };
-        })
-        .concat(
-          fieldsForOpenAPI(openAPI).filter(
-            (crdField) =>
-              providedAPI.specDescriptors &&
-              !providedAPI.specDescriptors.some((d) => d.path === crdField.path),
-          ),
-        )
-        // Associate `specDescriptors` with `fieldGroups` from OpenAPI
-        .map((field, i, allFields) =>
-          allFields.some((f) =>
-            f.capabilities.includes(
-              SpecCapability.fieldGroup.concat(
-                field.path.split('.')[0],
-              ) as SpecCapability.fieldGroup,
-            ),
-          )
-            ? {
-                ...field,
-                capabilities: [
-                  ...new Set(field.capabilities).add(
-                    SpecCapability.fieldGroup.concat(
-                      field.path.split('.')[0],
-                    ) as SpecCapability.fieldGroup,
-                  ),
-                ],
-              }
-            : field,
-        ),
-    [openAPI, providedAPI, clusterServiceVersion],
-  );
+  // Map providedAPI spec descriptors and openAPI spec properties to OperandField[] array
+  const initialFields: OperandField[] = React.useMemo(() => {
```

Comment:
Mostly just moving the logic into new helper functions. This memoized state still serves the same purpose as before.